src/invoice.rs

```diff
@@ -90,10 +94,26 @@ pub enum InvoiceParseError {
     /// invalid invoice.
     Invalid,
 
+    #[display(doc_comments)]
+    /// invalid invoice: contract ID with but no iface
+    ContractIdNoIface,
+
     #[display(doc_comments)]
     /// invalid contract ID.
     InvalidContractId(String),
 
+    #[display(doc_comments)]
+    /// invalid interface
+    InvalidIface(String),
+
+    #[display(doc_comments)]
+    /// invalid expiration timestamp
+    InvalidExpiration(String),
+
+    #[display(doc_comments)]
+    /// invalid query parameter
+    InvalidQueryParam,
```

Comment:
It will be good to proving the query parameter name in the error

Reply:
updated